### PR TITLE
Extension for SCRAM based authentication for Kafka connectivity. This includes SSL as well.

### PR DESCRIPTION
**Pull request by ritzshah**
Added Kafka extension with SCRAM authentication. You can also use SSL instead of SCRAM with this extension. 

This is tested and the image is located here : 
https://quay.io/repository/rshah/ansible-rulebook-kafka-scram-aio

The rule book example is located here : 
https://github.com/ritzshah/eda-rulebooks/blob/main/rulebooks/chat-survey.yml

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
Thank you very much for your contribution @ritzshah, but I think it is not justified to have a new plugin for kafka instead of update the existing one. Also it should have tests for the new options.

**Comment by ritzshah** [comment]
@Alex-Izquierdo 

Thanks for reviewing and your comments. I have tested this particular one in an openshift environment. I see that the existing plugin kafka.py is connecting to kafka, 
1) without user/password, 
2) SSL

Adding a 3rd option for SCRAM was becoming a little difficult in the same plugin and the plugin options were becoming more complex if someone wants to add all options in a rulebook, so I thought maybe like the file plugin kafka can have multiple plugin. 

Also can you clarify what do you mean b test options.

File: extensions/eda/plugins/event_source/kafka_scram_aio.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        bootstrap_servers=f"{host}:{port}",
+        group_id=group_id,
+        enable_auto_commit=True,
+        max_poll_records=1,
+        auto_offset_reset=offset,
+        security_protocol="SSL" if cafile else security_protocol,
+        ssl_context=context if cafile else None,
+        sasl_plain_password=sasl_plain_password,
+        sasl_plain_username=sasl_plain_username,
+        sasl_mechanism = None if cafile else sasl_mechanism
```
Comment (bzwei) [inline extensions/eda/plugins/event_source/kafka_scram_aio.py:105]: What is the sasl_mechanism if secuirty_protocol is SASL_SSL?

**Comment by bzwei** [comment]
Closing this one since #177 is merged.

**Pull request closed without merging**